x/ecocredit/server/testsuite/suite.go

```diff
@@ -618,45 +621,61 @@ func (s *IntegrationTestSuite) TestScenario() {
 		})
 	}
 
-	/****   TEST SET PRECISION   ****/
-	precisionCases := []struct {
-		name string
-		msg  ecocredit.MsgSetPrecision
-		ok   bool
+	/****   TEST CREDIT TYPES   ****/
+	creditTypeCases := []struct {
+		name        string
+		creditTypes []*ecocredit.CreditType
+		tx          ecocredit.MsgCreateClass
+		wantErr     bool
 	}{
 		{
-			"can NOT decrease the decimals", ecocredit.MsgSetPrecision{
-				Issuer: issuer1, BatchDenom: batchDenom, MaxDecimalPlaces: 2},
-			false,
-		}, {
-			"can NOT set to the same value", ecocredit.MsgSetPrecision{
-				Issuer: issuer1, BatchDenom: batchDenom, MaxDecimalPlaces: 7},
-			false,
-		}, {
-			"can increase", ecocredit.MsgSetPrecision{
-				Issuer: issuer1, BatchDenom: batchDenom, MaxDecimalPlaces: 8},
-			true,
-		}, {
-			"can NOT change precision of not existing denom", ecocredit.MsgSetPrecision{
-				Issuer: issuer1, BatchDenom: "not/existing", MaxDecimalPlaces: 1},
-			false,
+			name:        "valid eco credit creation",
+			creditTypes: []*ecocredit.CreditType{{Type: "carbon", Units: "kg", Precision: 3}},
+			tx: ecocredit.MsgCreateClass{Designer: s.signers[0].String(),
+				Issuers: []string{s.signers[1].String(), s.signers[2].String()}, Metadata: nil, CreditType: "carbon"},
```

Comment:
```suggestion
			tx: ecocredit.MsgCreateClass{
			        Designer: s.signers[0].String(),
				Issuers: []string{
				        s.signers[1].String(),
				        s.signers[2].String(),
				}, 
				Metadata: nil, 
				CreditType: "carbon",
		        },
```

Can you wrap these to 80cols and multiline format struct literals once they need wrapping?

Reply:
hows this look: https://github.com/regen-network/regen-ledger/pull/428/commits/dd9653f8a9ed26e82a6df79f5192d0096cf94f34 ?

Reply:
Bellissima!